libs/payments/cart/src/lib/cart.utils.ts

```diff
@@ -17,6 +19,35 @@ export const handleEligibilityStatusMap = {
   [EligibilityStatus.INVALID]: CartEligibilityStatus.INVALID,
 };
 
+export const CartEligibilityDetailsMap: Record<
+  EligibilityStatus,
+  CartEligibilityDetails
+> = {
+  [EligibilityStatus.CREATE]: {
+    eligibilityStatus: CartEligibilityStatus.CREATE,
+    state: CartState.START,
+  },
+  [EligibilityStatus.UPGRADE]: {
+    eligibilityStatus: CartEligibilityStatus.UPGRADE,
+    state: CartState.START,
+  },
+  [EligibilityStatus.DOWNGRADE]: {
+    eligibilityStatus: CartEligibilityStatus.DOWNGRADE,
+    state: CartState.FAIL,
+    errorReasonId: CartErrorReasonId.BASIC_ERROR,
+  },
+  [EligibilityStatus.BLOCKED_IAP]: {
+    eligibilityStatus: CartEligibilityStatus.BLOCKED_IAP,
+    state: CartState.FAIL,
+    errorReasonId: CartErrorReasonId.IAP_UPGRADE_CONTACT_SUPPORT,
+  },
+  [EligibilityStatus.INVALID]: {
+    eligibilityStatus: CartEligibilityStatus.INVALID,
+    state: CartState.FAIL,
+    errorReasonId: CartErrorReasonId.Unknown,
```

Comment:
Q: SHould this be uppercase `.UNKNOWN` to match `.BASIC_ERROR` above on L37 and `IAP_UPGRADE_CONTACT_SUPPORT` on L42?

Reply:
@pdehaan Good question. @julianpoy should I go ahead and update for consistency? Or is there a reason it is not upper-cased as the other two?

Reply:
All of these should be PascalCase